extensions/http-curl/CMakeLists.txt

```diff
@@ -42,9 +42,9 @@ if(CMAKE_THREAD_LIBS_INIT)
 endif()
 
 if (CURL_FOUND)
-        include_directories(${CURL_INCLUDE_DIRS})
-        target_link_libraries (minifi-http-curl ${CURL_LIBRARIES})
-endif(CURL_FOUND)
+  include_directories(${CURL_INCLUDE_DIRS})
+  target_link_libraries(minifi-http-curl ${CURL_LIBRARIES})
```

Comment:
^ Even in the case of a system library we could rely on the static library if it exists. We don't do it now to ensure we have a small binary size, but we should make it clear that while downloading the source and building manually is an option in most cases a static curl lib likely already exists on the build system. In my case I've just checked my centos 7,6, u16-18, fedora, and suse VMs and they all have static libs. There should be a little work here to ensure we get that IF the user wants that. Perhaps that necessitates an added option?

Reply:
Agreed that we should have the option to static link when not using bundled. I think that should be done in a new ticket that covers all dynamic libs that have static linking options when linking system libs, possibly with a general option e.g. STATIC_LINK_ALL.